### PR TITLE
Various fixes regarding RFC-8621 Distributed tests

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/UploadRoutes.scala [hunk 1]
```diff
@@ -51,6 +51,8 @@ import reactor.core.scala.publisher.SMono
 import reactor.core.scheduler.Schedulers
 import reactor.netty.http.server.{HttpServerRequest, HttpServerResponse}
 
+case class TooBogUploadException() extends RuntimeException
```
Comment (LanKhuat) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/UploadRoutes.scala:54]: s/Bog/Big

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java [hunk 2]
```diff
@@ -826,6 +826,14 @@ void headerContainsShouldReturnUidsOfMessageHavingThisHeaderWithTheSpecifiedValu
             .containsOnly(m1.getUid(), m6.getUid(), m8.getUid(), m9.getUid());
     }
 
+    @Test
+    void headerContainsShouldBeCaseIncentive() throws Exception {
```
Comment (LanKhuat) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java:830]: s/Incentive/Insensitive

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala [hunk 3]
```diff
@@ -4662,6 +4662,65 @@ trait EmailQueryMethodContract {
...
+
+      assertThatJson(response)
+        .withOptions(new Options(IGNORING_ARRAY_ORDER))
+        .inPath("$.methodResponses[0][1].ids")
+        .isEqualTo(s"""["${messageId1.serialize}"]""")
+    }
+  }
+
+  @Test
+  def subjectShouldBeCaseIncentive(server: GuiceJamesServer): Unit = {
```
Comment (LanKhuat) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala:4723]: s/Incentive/Insensitive

**Review by LanKhuat — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**